### PR TITLE
Mdc migration chip, button, menu, table, checkbox and list

### LINKED ISSUE #2934: Angular material mdc migration
With angular 15 there was an update for the angular material components. 

Currently we are using `MatLegacy...` modules which should be replaced in the near future with the new MDC components. 

There is a migration tool which automatically replaces the used modules and updates some other things. However this needs testing and manual intervention in at least most cases. For that the migration tool has a feature to do this migration only for specific modules (which has the drawback that the main `styles.scss` will not be modified). 

A migration guide can be found here: https://material.angular.io/guide/mdc-migration

Components to migrate
- [x] Autocomplete
- [ ] Button #3703 
- [x] Card #3487
- [ ] Checkbox #3696 
- [ ] Chips
- [x] Dialog #3397 
- [x] Form Field #3002
- [x] Input #3002
- [ ] List
- [ ] Menu
- [x] Option / Optgroup #3002
- [x] Paginator
- [x] Progress Bar #3097
- [x] Progress Spinner #3097
- [x] Radio #3001
- [x] Select #3002
- [x] Slide Toggle
- [x] Slider #2998
- [x] Snack Bar #3097
- [ ] Table #3694 
- [x] Tabs #2998
- [x] Tooltip #2999

### PR DESCRIPTION
**Pull request by bastianjoel**
resolves #2934
closes #3694
closes #3696 
closes #3703

### CODE REVIEW

**Comment by bastianjoel** [comment]
We should maybe replace the chips with an custom component. 
@Elblinator maybe you could have a look on that?

**Comment by bastianjoel** [comment]
@MSoeb Did you already discuss with @Elblinator that the change request in https://github.com/OpenSlides/openslides-client/pull/3694 and came to the conclusion that we will not be fix them?

**Review by Elblinator — CHANGES_REQUESTED** [review]
Please merge into main, a PR is missing which makes the projection mainly not testable

Already found change requests:

- [x] Imports: if a file is selected then the file-name is lower than the button ‘select file’ and the clear button
Screenshot_20240607_141602
- [x] committee-detail: mat-card-header from a meeting→the header is a bit too tall (it was height:40px plus padding-top:6px; now its ~48px plus padding-top:6
 probably bc of the wrong height: title too far up, “archived” and “public template” indicator too far down
Screenshot_20240607_142804
- [ ] Account-create →the two buttons in the Initial password do not fit into the mat-form-field; please make them fit into one line again;
Screenshot_20240607_143105
- [x] Design edit/create dialogue → the colour field are now too big , please make them fit in a line again (max per line = 3)
- [x] right corner > my profile > change password: Error in console and not input fields: ERROR Error: NG0300: Multiple components match node with tagname button: MatButton and MatIconButton. Find more at https://angular.io/errors/NG0300
    at throwMultipleComponentError (core.mjs:11607:11)
- [ ] LoS- detail-view > three dot menu: the icons and the text are not aligned, please align them
Screenshot_20240607_143456
- [ ] projector-detail > current LoS: projector buttons have an ugly shadow
- [ ] meeting settings > setting detail: distance between items is very big: please remove the class setting-form-group 
- [ ] Motion-detail-view: The icons for a projected motion and the star-icon are differently aligned:
Screenshot_20240607_142916
- [x]  in Chromium: [not always reproducable] if you open a dialog then a scrollbar appears
- [ ] [not always reproducable] Filter bar: if the option mat-divider and another checkbox is present then there is a unnecessary scrollbar-> This could be due to delayed loading. The scrollbar was there for the first few moments, but after a while it disappeared completely


Only notes for now: the following things need further discussion before they can be fixed:
they will be discussed after the change requests are implemented,
please do not fix them yet
- Nav-toolbar-content: on orga level and on meeting level look different, They should be unified. _Given that a lot of places are now in the darker theme: I would suggest turning the toolbar on the meeting-level darker_
- Group-Matrix give permissions more space the mat-cell is ~15px smaller and there is now padding-right:16px. Can the width be again 15 px bigger. I am unsure whether or not the paddig should stay there or not

**Comment by bastianjoel** [comment]
> in Chromium: [not always reproducable] if you open a dialog then a scrollbar appears

While this might be related to another MDC migration this is not related to this PR. Please open another issue.

**Comment by bastianjoel** [comment]
> [not always reproducable] Filter bar: if the option mat-divider and another checkbox is present then there is a unnecessary scrollbar-> This could be due to delayed loading. The scrollbar was there for the first few moments, but after a while it disappeared completely

As discussed

**Review by Elblinator — APPROVED** [review]
several places changed the appearance, but this was expected
In my opinion the appearance is okay as it is. 
But please @MSoeb also test it and give your feedback
We can of course also test together to get a better picture

**Review by Elblinator — CHANGES_REQUESTED** [review]
committee-detail if the meeting can forward or can be forwarded to by several other meetings:
the "More"/"Less" and the icon expand_more/expand_less is not aligned anymore

**Review by Elblinator — CHANGES_REQUESTED** [review]
A not complete list (testing everything will take a moment longer)
Please change the following things back to how they were before or state if it's the new default or hard to change back/why it's hard

- [x] In dashboard, the icon "layers" with the tooltip "Show committee" is not blue, please return it to the old colour
- [x] in every three-dot menu: 
    - [x] the text changed from font-size:14 to font-size : 16
    - [x] the icon changed from color: rgba(0, 0, 0, 0.54) to rgba(0, 0, 0, 0.87)
- [x] in  every list view the sort and filter button (icon and text) changed from color: rgba(0, 0, 0, 0.87) to color: rgba(0, 0, 0)
- [x] in the sort thing. The arrow (icon "arrow_downward"/"arrow_upward") changed the color from rgba(0, 0, 0, 0.54) to rgba(0, 0, 0, 0.87)
- [x] In the Filter-list mat-panel-title 
   - [x] font-size from 15px to 14px
   - [x] font-weight: from 400 to 500
- [x] as far as we could see every text (input/label) in a mat-form field changed it's font-size from 15px to 16px
- [ ] buttons:
   - [x] letter spacing changed from normal to 1.25
   - [ ] the color change from white to black (vise versa dependent on the dark/light mode)
   - [ ] mat-checkbox (e.g. in account edit view) the little check seems to be less central  
   - [ ] with mat-icon: changed size from 40 x40 to 48x48
- [ ] orga-level: tags font-weight of 500 is gone
- [ ] os-file-list home icon got smaller, (font-size changed from 24 to 18)
- [ ] in a dialog > mat-dialog-content > form > p:
   - [ ] font-size: 15px to 16px
   - [x] letter-spacing: normal to 0.5px 
 - [ ] mat-panel-title: 
    - [ ]  font-size: 15px to 14px
    - [ ] font-weight: 400 to 500
 - [ ] mat-list-item: letter-spacing from normal to 0,5

**Comment by bastianjoel** [comment]
@MSoeb @Elblinator I am now finished working through our internal list with change requests. 

Everything marked there as wontfix are points which would require additional effort to implement or come with drawbacks in maintainability. If any of those points are mandatory we should discuss them.

**Review by Elblinator — COMMENTED** [review]
- [ ] In History: The refresh button is too big if the language is changed to German

- [ ] Checkboxes have different inside colours: 

- some change colour together with light and dark-mode (e.g. top-right-corner your own present box)
- some stay white in the middle
places I found these checkboxes:  meetings> settings, projector-list > edit projector dialog, participant-list, participant-detail-edit-view, structure-level create dialog, election-create-view, motion-detail.... 
If needed I can try to create the  full list, but for now I'll stop at this point of the list

**Comment by bastianjoel** [comment]
> Checkboxes have different inside colours

I did this on purpose because I thought the contrast is quite bad and did not notice that it was black previously. The presence button uses a transparent icon and I cannot change it easily there. Should I change it back to black or is it okay as it is not?

**Review by Elblinator — APPROVED** [review]

File: client/src/app/site/pages/meetings/pages/projectors/modules/projector-detail/components/projector-detail/projector-detail.component.html [hunk 1]
```diff
@@ -91,7 +91,9 @@ <h2>{{ projector.name }}</h2>
                         <mat-icon>refresh</mat-icon>
                     </button>
                     <!-- scaling indicator -->
-                    <div class="button-size" [ngClass]="projector.scale !== 0 ? 'warn' : ''">{{ projector.scale }}</div>
+                    <span class="button-size" [ngClass]="projector.scale !== 0 ? 'warn' : ''">
+                        {{ projector.scale }}
+                    </span>
```
Comment (reiterl) [inline client/src/app/site/pages/meetings/pages/projectors/modules/projector-detail/components/projector-detail/projector-detail.component.html:96]: class ngClass mix

File: client/src/app/site/pages/meetings/pages/projectors/modules/projector-detail/components/projector-detail/projector-detail.component.html [hunk 2]
```diff
@@ -142,32 +144,32 @@ <h2>{{ projector.name }}</h2>
                         <mat-icon>refresh</mat-icon>
                     </button>
                     <!-- scroll indicator -->
-                    <div class="button-size" [ngClass]="projector.scroll !== 0 ? 'warn' : ''">
+                    <span class="button-size" [ngClass]="projector.scroll !== 0 ? 'warn' : ''">
```
Comment (reiterl) [inline client/src/app/site/pages/meetings/pages/projectors/modules/projector-detail/components/projector-detail/projector-detail.component.html:147]: same.

**Review by reiterl — COMMENTED** [review]
I am not sure, if it is an issue, but I found some location in the source code, where:
class, ngClass and [class] is mixed. Else look good.

File: client/src/app/site/modules/chip-select/components/chip-select/chip-select.component.html [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-        [matMenuTriggerFor]="canOpen ? menu : null"
-        [ngClass]="{ 'menu-handle': canOpen }"
-    >
-        <mat-basic-chip *ngFor="let chip of chips" [class]="chipClass || 'grey'" disableRipple>
+    <mat-chip-listbox class="mat-chip-list-stacked" [matMenuTriggerFor]="canOpen ? menu : null">
+        <mat-basic-chip
+            class="chip-stacked"
+            *ngFor="let chip of chips"
+            [ngClass]="{ 'menu-handle': canOpen }"
+            [class]="chipClass || 'grey'"
```
[the remark this answers is not in the document]
Comment (bastianjoel) [reply client/src/app/site/modules/chip-select/components/chip-select/chip-select.component.html:7]: This is valid

**Pull request merged**